examples/mongocxx/get_values_from_documents.cpp

```diff
@@ -34,44 +49,47 @@ using namespace bsoncxx;
 //     ]
 // }
 
-// Construct a document in the format of 'messagelist'
-document::value new_message(int64_t uid, int32_t status, std::string msg) {
-    builder::stream::document doc;
-    return doc << "uid" << uid << "status" << status << "msg" << msg << builder::stream::finalize;
+// Construct a document in the format of 'messagelist'.
+doc_value new_message(int64_t uid, int32_t status, std::string msg) {
+    doc_builder doc;
+    return doc << "uid" << uid << "status" << status << "msg" << msg << finalize;
 }
 
-// Insert a document into the database
+// Insert a document into the database.
 void insert_test_data(mongocxx::collection& coll) {
-    builder::stream::document builder{};
-    builder << "messagelist" << builder::stream::open_array
+    doc_builder builder{};
+    builder << "messagelist" << open_array
             << new_message(413098706, 3, "Lorem ipsum...")
             << new_message(413098707, 2, "Lorem ipsum...")
-            << new_message(413098708, 1, "Lorem ipsum...") << builder::stream::close_array;
+            << new_message(413098708, 1, "Lorem ipsum...") << close_array;
 
-    document::value doc = builder << builder::stream::finalize;
+    doc_value doc = builder << finalize;
 
-    // Normally, one should check the return value for success
+    // Normally, one should check the return value for success.
     coll.insert_one(doc.view());
 }
 
-// Iterate over contents of messagelist
-void iterate_messagelist(document::element& ele) {
-    // Check validity && type before trying to iterate
-    if (ele.type() == type::k_array) {
-        array::view subarray{ele.get_array().value};
-        for (array::element msg : subarray) {
+// Iterate over contents of messagelist.
+void iterate_messagelist(const doc_element& ele) {
+    // Check validity and type before trying to iterate.
+    if (ele.type() == k_array) {
+        array_view subarray{ele.get_array().value};
+        for (array_element& msg : subarray) {
```

Comment:
Nit: suggest making `msg` a non-reference type (`element` is only 16 bytes and is trivially destructible) or a const reference, if you don't need to modify it.